src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/EventIdValidator.java

```diff
@@ -0,0 +1,37 @@
+package uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd;
+
+import io.vavr.control.Validation;
+
+import java.util.function.Function;
+import javax.annotation.Nonnull;
+
+import static io.vavr.control.Validation.invalid;
+import static io.vavr.control.Validation.valid;
+import static java.lang.String.format;
+
+public final class EventIdValidator {
+
+    private static final String EVENT_ID_ATTACH_TO_CASE = "attachToExistingCase";
+    private static final String EVENT_ID_CREATE_CASE = "createCase";
+
+    private EventIdValidator() {
+        // utility class constructor
+    }
+
+    @Nonnull
+    static Validation<String, Void> isAttachToCaseEvent(String eventId) {
+        return hasValidEventId(EVENT_ID_ATTACH_TO_CASE::equalsIgnoreCase, eventId);
+    }
+
+    @Nonnull
+    static Validation<String, Void> isCreateCaseEvent(String eventId) {
+        return hasValidEventId(EVENT_ID_CREATE_CASE::equalsIgnoreCase, eventId);
```

Comment:
can't ignore case for this check :)

Reply:
I know it exists currently but it looks like a bug

Reply:
lol, breaking change compared to previous implementation. at least for attach to case event